cdbsearch.py

```diff
@@ -100,6 +103,32 @@ def __apicall(self, url, timeout):
         self.count_inflightRequests.dec()
         return content
 
+    def add_cdb_pv_positions(self, epd):
+        """query cdb for the PV of the position and create a dictionary containing these positions and their distance to the PV leaf for extensions during search"""
+        api = "http://www.chessdb.cn/cdb.php"
+        timeout = 15
+        url = api + f"?action=querypv&board={epd}&json=1"
+        content = self.__apicall(url, timeout)
+        pvlen = 0
+        if (
+            content
+            and "status" in content
+            and content["status"] == "ok"
+            and "pv" in content
+        ):
+            pv = content["pv"]
+            pvlen = remaining = len(pv)
+            board = chess.Board(epd)
+            self.cdbPvToLeaf[board.epd()] = remaining
+            self.executorWork.submit(self.queryall, board.epd())
+            for m in pv:
+                move = chess.Move.from_uci(m)
+                board.push(move)
+                remaining -= 1
+                self.cdbPvToLeaf[board.epd()] = remaining
+                self.executorWork.submit(self.queryall, board.epd())
```

Comment:
So in my cleanup I stumbled upon this line: The skipTT argument here seems to be missing. My plan is to make skipTT=False the default, but I do not understand how the code could run as it is? Does `submit` populate missing arguments with zeros or something? In that case, the code would work as intended.

Reply:
I agree this needs fixing, even if there is some default behavior. I suspect it might be reading some random data.